src/main/java/com/epam/emotionalhelp/service/impl/EmotionServiceImpl.java

```diff
@@ -1,48 +1,58 @@
 package com.epam.emotionalhelp.service.impl;
 
 import com.epam.emotionalhelp.controller.dto.EmotionRequestDto;
-import com.epam.emotionalhelp.mapper.EmotionMapper;
-import com.epam.emotionalhelp.model.Emotion;
+import com.epam.emotionalhelp.controller.dto.EmotionResponseDto;
+import com.epam.emotionalhelp.controller.response.ResponseMessage;
+import com.epam.emotionalhelp.exceptionhandler.exception.ResourceNotFoundException;
 import com.epam.emotionalhelp.repository.EmotionRepository;
 import com.epam.emotionalhelp.service.EmotionService;
+import com.epam.emotionalhelp.service.mapper.EmotionMapper;
 import lombok.RequiredArgsConstructor;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
 @RequiredArgsConstructor
 public class EmotionServiceImpl implements EmotionService {
     private final EmotionRepository emotionRepository;
 
     @Override
-    public Page<Emotion> findAll(Pageable pageable) {
-        return emotionRepository.findAll(pageable);
+    public Page<EmotionResponseDto> findAll(Pageable pageable) {
+        return EmotionMapper.pageEntityToPageDto(emotionRepository.findAll(pageable));
     }
 
     @Override
-    public Emotion addQuestion(EmotionRequestDto emotionRequestDto) {
-        return emotionRepository.save(EmotionMapper.toEntity(emotionRequestDto));
+    public EmotionResponseDto findById(Long id) {
+        return EmotionMapper.toDto(emotionRepository.findById(id).orElseThrow(() ->
```

Comment:
We have duplicated code in this class. Please extract that code into private method and refactor service methods.